### ISSUE TITLE
Examples directory contains a single trivial script; no cookbook

### ISSUE THREAD

**Issue by sachncs** [sachncs/avqa#25]
## Problem

The project ships exactly one example, `examples/01_basic_attention.py` (53 lines), which constructs an `AVQAttention` module on random tensors and prints the output shape. There is no:

- Causal-masked decoding example (the canonical use case for KV-cache attention).
- Multi-pass refinement example.
- HVAQ vs paper-equivalent comparison.
- BCAR codebook adaptation walkthrough.
- Profiler + Visualizer usage example.
- Hugging Face / vLLM / Triton adapter example (CHANGELOG records these as removed).
- Numerical equivalence test against `torch.nn.functional.scaled_dot_product_attention` (the natural baseline).

## Evidence

`ls examples/`:

```
01_basic_attention.py
```

`examples/01_basic_attention.py:42-49`:

```python
query = torch.randn(batch_size, seq_len, embed_dim)
key = torch.randn(batch_size, seq_len, embed_dim)
value = torch.randn(batch_size, seq_len, embed_dim)
output = attention(query, key, value)
print(f"Input  shape: {tuple(query.shape)}")
print(f"Output shape: {tuple(output.shape)}")
print(f"Sample output norm: {output.norm(dim=-1).mean().item():.4f}")
```

`README.md` Quick Start (lines 73-107) shows the module and functional APIs inline; the README does not point to any examples in `examples/` (no "Examples" link in the Quick Start).

## Root Cause

The examples directory was scaffolded for the v0.1.0 release but only the most basic case was filled in.

## Impact

- A user cannot see how AVQA handles causal decoding, KV caching, or HVAQ/BCAR without reading the source.
- The README "Quick Start" demonstrates construction but not usage; the gap between "you constructed a module" and "you used it on real data" is undocumented.
- Examples are a primary trust signal for OSS visitors; a directory with one trivial file undercuts that.

## Recommendation

Add at least five runnable examples that exercise the major API surfaces:

1. `02_causal_decoding.py` — incremental decoding with `InMemoryKVCache`, demonstrating `kv_cache.append` / `kv_cache.lookup` and the autoregressive loop.
2. `03_hvaq_vs_paper.py` — same forward pass with `HopfieldConfig(adaptive="none")` vs `adaptive="entropy"` and a numerical-equivalence report.
4. `04_bcar_adaptation.py` — train a tiny codebook on a synthetic stream and visualize mean-constraint preservation.
5. `05_profiler_visualizer.py` — run a profiled forward pass, export JSON, and render a refinement tree.
6. `06_vs_sdpa.py` — head-to-head latency + numerical-equivalence comparison against `torch.nn.functional.scaled_dot_product_attention` (uses `benchmarks/repro_cpu.py` as the source of truth).

Each example should be `< 80` lines, runnable with `python examples/NN_*.py`, and produce stdout output that demonstrates value (numerical comparison, KV-cache hit rate, etc.).

## Acceptance Criteria

- [ ] `examples/` contains at least 5 runnable scripts (excluding `01_basic_attention.py`).
- [ ] Each new example exits 0 on a clean checkout.
- [ ] README Quick Start links to the relevant examples ("See `examples/02_causal_decoding.py` for autoregressive usage").

## Verification

Run `python examples/0N_*.py` for each new example; confirm exit code 0 and that the printed output is informative.

## Risk

Low. Examples are additive documentation; no source changes.

**Comment by sachncs** [comment]
Resolved in PR #37.
